sample/src/main/java/org/opennms/integration/api/sample/MyHealthCheck.java

```diff
@@ -106,20 +118,53 @@ public Response perform(Context context) throws Exception {
             } catch (Exception e) {
                 return new ResponseBean(e);
             }
+            // Sample Poller Health Check
             try {
                 CompletableFuture<PollerResult> pollerStatus = pollerClient.poll()
                         .withAddress(InetAddress.getLocalHost())
                         .withPollerClassName(SamplePoller.class.getCanonicalName())
                         .withServiceName("Sample")
                         .execute();
-                if (pollerStatus.get().getStatus().equals(org.opennms.integration.api.v1.pollers.Status.Up)) {
-                    return new ResponseBean(Status.Success);
-                } else {
+                if (!pollerStatus.get().getStatus().equals(org.opennms.integration.api.v1.pollers.Status.Up)) {
                     return new ResponseBean(Status.Failure, pollerStatus.get().getReason());
                 }
             } catch (Exception e) {
                 return new ResponseBean(e);
             }
+
+            // Sample Collector Health Check
+            int nodeId = getFirstNode();
+            // If there is node, can't check collector, return success as all previous checks succeeded.
+            if(nodeId <= 0) {
+                return new ResponseBean(Status.Success);
+            }
+            try {
+                CompletableFuture<CollectionSet> collectionSetFuture = collectorClient.collect()
+                        .withCollectorClassName(SampleCollector.class.getCanonicalName())
+                        .withRequest(new SampleCollector.CollectionRequestImpl(nodeId))
+                        .execute();
+                CollectionSet collectionResult = collectionSetFuture.get();
+                if(collectionResult.getStatus().equals(CollectionSet.Status.SUCCEEDED)) {
+                    if(SampleCollector.validateCollectionSet(collectionResult)) {
+                        return new ResponseBean(Status.Success);
+                    } else {
+                        return new ResponseBean(Status.Failure, "collection set didn't match");
+                    }
+                } else {
+                    return new ResponseBean(Status.Failure, "Sample Collector Collection Failed");
+                }
+            } catch (Exception e) {
+                return new ResponseBean(e);
+            }
+        }
+    }
+
+    private int getFirstNode() {
+        int nodeId = 0;
+        if(nodeDao.getNodeCount() > 0) {
+            List<Integer> nodeIds = nodeDao.getNodeIds();
```

Comment:
We can avoid the call to `getNodeCount()`, and simply call `getNodeIds()`, if the list is empty, there are no nodes.